### PR TITLE
Remove SweetAlert

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
### Summary

- Remove SweetAlert2 dependencies 
  - Replace it with a custom Popup
  - Add typescript support notations for some global variables 

### Will affect visual aspect of the product

YES

### Screenshots 

image

image

### Test instructions

Test the popup when:
 - Deleting a field group.
 - Deleting input fields when no field is selected.

## Check before Pull Request is ready:

* [ ] I have [written a test](CONTRIBUTING.md#writing-an-acceptance-test) and included it in this PR
* [x] I have [run all tests](CONTRIBUTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](CONTRIBUTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](CONTRIBUTING.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](CONTRIBUTING.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

Closes https://github.com/Codeinwp/woocommerce-product-addon/issues/340

### CODE REVIEW

File: js/admin/ppom-meta-table.js [hunk 1]
```diff
@@ -52,12 +44,22 @@ jQuery(function($){
...
-						swal.fire(resp, "", "error").then();
+						window?.ppomPopup?.open({
+							title: window?.ppom_vars?.i18n.popup.finishTitle,
+							hideCloseBtn: true,
+							onConfirmation: () => location.reload(),
+							onClose: () => location.reload()
+						});
+					} else {
+						window?.ppomPopup?.open({
+							title: window.ppom_vars.i18n.popup.errorTitle,
```
Comment (carlalexander) [inline js/admin/ppom-meta-table.js:55]: You're not using `window?.ppom_vars?` here. Any reason why?

**Review by carlalexander — COMMENTED** [review]
I don't have much to say for this PR! Looks fine. I have a few questions.

Why are naming `ppomPopup` using camel case when everything else uses snake case. Shouldn't we do `ppom_popup`?

My other question relates to the use of `window?.ppomPopup?` and `window?.ppom_vars?`. Is there are reason why we're using the optional chaining? Wouldn't we want to know if `ppomPopup` or `ppom_vars` are missing instead of ignoring it?

Finally, any way we can write some e2e test to test the change?

**Review by abaicus — COMMENTED** [review]
I agree with @carlalexander here. Is there any chance that ppom_vars or the class class instance will be `undefined`? IMO, the `i18n.popup` key should be there as long as it's used in the script. Without these being defined, the intended behavior will not be working, so instead of using optional chaining, I would rather make sure that those variables are defined. 

More than that, the popup should probably be requested as a dependency of both scripts ([here it is not](https://github.com/Codeinwp/woocommerce-product-addon/pull/342/files#diff-b0dd880f7d63a018e89f3576fda07c2035846507b85e89a9ace2e2bee563a8b5R72)).

---

Regarding the reasoning behind using camelCase combined with snake_case, I think that's how the script was previously localized, so it would mean changing that reference everywhere. I personally prefer using camelCase in JS usually, but I understand the difference in this specific case. 

I think we should strive for one or the other in the future.

File: js/admin/ppom-meta-table.js (hunk 1 above)
Comment (Soare-Robert-Daniel) [reply js/admin/ppom-meta-table.js:55]: WordPress will not load the script unless the global object is also loaded (in this case, `ppom_vars`).

**Comment by carlalexander** [comment]
> Regarding the reasoning behind using camelCase combined with snake_case, I think that's how the script was previously localized, so it would mean changing that reference everywhere. I personally prefer using camelCase in JS usually, but I understand the difference in this specific case.
> 
> I think we should strive for one or the other in the future.

Yeah, that was my main question. I just want to standardize on one type. 👍

**Review by carlalexander — APPROVED** [review]
Don't have much else to add besides my existing questions. If @abaicus feels everything is good, we're good 😃

**Review by abaicus — APPROVED** [review]

**Pull request merged**